plugin/pkg/admission/noderestriction/admission.go

```diff
@@ -192,18 +194,30 @@ func (c *nodePlugin) admitPodStatus(nodeName string, a admission.Attributes) err
 
 func (c *nodePlugin) admitNode(nodeName string, a admission.Attributes) error {
 	requestedName := a.GetName()
-
-	// On create, get name from new object if unset in admission
-	if len(requestedName) == 0 && a.GetOperation() == admission.Create {
+	if a.GetOperation() == admission.Create {
 		node, ok := a.GetObject().(*api.Node)
 		if !ok {
 			return admission.NewForbidden(a, fmt.Errorf("unexpected type %T", a.GetObject()))
 		}
-		requestedName = node.Name
+		// On create, get name from new object if unset in admission
+		if len(requestedName) == 0 {
+			requestedName = node.Name
+		}
 	}
-
 	if requestedName != nodeName {
 		return admission.NewForbidden(a, fmt.Errorf("node %s cannot modify node %s", nodeName, requestedName))
 	}
+
+	if a.GetOperation() == admission.Create || a.GetOperation() == admission.Update {
+		node, ok := a.GetObject().(*api.Node)
+		if !ok {
+			return admission.NewForbidden(a, fmt.Errorf("unexpected type %T", a.GetObject()))
+		}
+		// On create or update, only allow Node.Spec.ConfigSource to be set if the DynamicKubeletConfig feature gate is enabled
```

Comment:
actually, what I meant was, don't let a Node create its API object referencing a config map, or update its spec to add a reference to a configmap (compare a.GetObject() and a.GetOldObject())